modules/scylladb/scylladb_test.go

```diff
@@ -0,0 +1,194 @@
+package scylladb_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	smithyendpoints "github.com/aws/smithy-go/endpoints"
+	"github.com/gocql/gocql"
+	"github.com/stretchr/testify/require"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/scylladb"
+)
+
+func TestScyllaDB(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+		scylladb.WithShardAwareness(),
+	)
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+
+	t.Run("test without shard awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 19042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var address string
+		err = session.Query("SELECT address FROM system.clients").Scan(&address)
+		require.NoError(t, err)
+	})
+
+	t.Run("test with shard awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 19042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var address string
+		err = session.Query("SELECT address FROM system.clients").Scan(&address)
+		require.NoError(t, err)
+	})
+}
+
+func TestScyllaWithConfigFile(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+		scylladb.WithConfigFile(filepath.Join("testdata", "scylla.yaml")),
+		scylladb.WithShardAwareness(),
+	)
+	require.NoError(t, err)
+	testcontainers.CleanupContainer(t, ctr)
+
+	t.Run("test without shard awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 9042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var cluster_name string
+		err = session.Query("SELECT cluster_name FROM system.local").Scan(&cluster_name)
+		require.NoError(t, err)
+
+		require.Equal(t, "Amazing ScyllaDB Test", cluster_name)
+	})
+
+	t.Run("test with shard awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 19042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var cluster_name string
+		err = session.Query("SELECT cluster_name FROM system.local").Scan(&cluster_name)
+		require.NoError(t, err)
+
+		require.Equal(t, "Amazing ScyllaDB Test", cluster_name)
+	})
+}
+
+func TestScyllaWithAlternator(t *testing.T) {
```

Comment:
question: These next two tests look like duplicates of the above, or could be combined?